infra/Pos/NtpCheck.hs

```diff
@@ -0,0 +1,65 @@
+{-# LANGUAGE ScopedTypeVariables #-}
+
+module Pos.NtpCheck
+    ( withNtpCheck
+    , NtpStatus(..)
+    ) where
+
+import           Universum                   hiding (bracket)
+
+import           Control.Monad.Trans.Control (MonadBaseControl)
+import qualified Data.List.NonEmpty          as NE
+import           Data.Time.Units             (Microsecond)
+import           Mockable                    (Bracket, CurrentTime, Fork, Mockable,
+                                              bracket, currentTime)
+import           NTP.Client                  (NtpClientSettings (..), pressNtpStopButton,
+                                              startNtpClient)
+import           Pos.Core.Timestamp          (Timestamp (..), diffTimestamp)
+import           Pos.Infra.Configuration     (HasInfraConfiguration, infraConfiguration)
+import qualified Pos.Infra.Configuration     as Infra
+import           Pos.Util.Util               (median)
+import           Serokell.Util               (sec)
+import           System.Wlog                 (WithLogger)
+
+type NtpCheckMonad m =
+    ( MonadIO m
+    , MonadMask m
+    , MonadBaseControl IO m
+    , Mockable Fork m
+    , Mockable Bracket m
+    , Mockable CurrentTime m
+    , WithLogger m
+    , HasInfraConfiguration
+    )
+
+withNtpCheck :: forall m a. NtpCheckMonad m => (NtpStatus -> m ()) -> m a -> m a
+withNtpCheck onStatus action = bracket (startNtpClient ntpSettings) pressNtpStopButton (\_ -> action)
```

Comment:
Ah, I see you don't want it to be single shot, but rather want it to run until it fails.
I think you should just use `startNtpServer`, or better fix `startNtpClient` to avoid `fork`s (if you want to enter `node-sketch territory 😝 ).

Reply:
`pressNtpStopButton` should be used when you really want to stop it in normal case. In our case we never want to stop it, except when we stop the whole program.
Clean-up in case of exceptions should be handled by ntp client itself, imo.

Reply:
I tested this code and it appears to work. Although it's unfortunate that it doesn't use `async`, I don't know how this code behaves in case of an exception in `node-sketch` (but I don't want to touch `node-sketch` in this task).

Reply:
Did you wait for 4 hours or test it with smaller interval (i. e. did you test that it's triggered multiple times, not only on start)?

Reply:
I tested with 5s interval

Reply:
Ok, I see, nevermind, didn't read it carefully.